Parsers/ASimNetworkSession/Parsers/vimNetworkSessionCheckpointFirewall.yaml

```diff
@@ -0,0 +1,169 @@
+                              array_length(src_or_any) == 0 and array_length(dst_or_any) == 0, "-", // match not requested
+                              (temp_isSrcMatch and temp_isDstMatch), "Both", // has to be checked before the individual 
+                              temp_isSrcMatch, "SrcIpAddr",
+                              temp_isDstMatch, "DstIpAddr",
+                              "No match"
+                          )
+          | where ASimMatchingIpAddr != "No match"
+          | extend 
+                  EventVendor = DeviceVendor,
+                  EventProduct = "VPN & Firewall",
+                  EventCount = toint(1),
+                  EventType = "NetworkSession",
+                  EventOriginalType = "TRAFFIC",
+                  EventSchema = "NetworkSession",
+                  EventSchemaVersion = "0.2.3",
+                  DeviceAction = coalesce(DeviceAction, extract(@'rule_action=(\w+)',1,AdditionalExtensions, typeof(string))),        
+                  NetworkRuleName = extract(@'rule_uid=([\w+\-]*)',1,AdditionalExtensions, typeof(string)),
+                  EventOriginalUid = extract(@'loguid=(\{[^\}]+\})',1,AdditionalExtensions, typeof(string)),
+                  Dvc = coalesce(extract(@'origin=([\d+\.]*)',1,AdditionalExtensions, typeof(string)), Computer),
+                  DvcHostname = extract(@'originsicname=CN\\=(.+)\,O',1,AdditionalExtensions, typeof(string)),
+                  SrcZone = extract(@'inzone=([^;]+)',1,AdditionalExtensions, typeof(string)),
+                  DstZone = extract(@'outzone=([^;]+)',1,AdditionalExtensions, typeof(string)),
+                  CommunicationDirection = extract(@'conn_direction=(\w+)',1,AdditionalExtensions, typeof(string)),                  
+                  DstAppName = coalesce(Activity, extract(@'service_id=(\w+)',1,AdditionalExtensions, typeof(string))),
+                  ThreatCategory = case (
+                                    AdditionalExtensions has "alert=", extract(@'alert=([\w+\-]*)',1,AdditionalExtensions, typeof(string)),
+                                    AdditionalExtensions has "inspection_category=", extract(@'inspection_category=([\w+\-]*)',1,AdditionalExtensions, typeof(string)),
+                                    ""),
+                  ThreatName = extract(@'inspection_item=([\w+\-]*)',1,AdditionalExtensions, typeof(string))
+          | extend
+                  EventResult = case(
+                                      isempty(DeviceAction), "", 
+                                      DeviceAction == "Accept", "Success",
+                                      "Failure"
+                                  ), 
+                  NetworkRuleName = coalesce(DeviceCustomString2, NetworkRuleName),
+                  Protocol = toint(Protocol),
+                  EventStartTime = unixtime_milliseconds_todatetime(tolong(ReceiptTime))
+          | where ((eventresult == "*") or (EventResult == eventresult))
```

Comment:
Two strategies to pre-filter on EvenResult:
- Hard: Prefilter on AdditionalExtensions (where eventresult == "Success" and AddionalExtention has "Accept" etc.)
- Simpler: just calculate EventResult first thing (line 84 or so) and filer by it.